core/AtlasState.cpp

```diff
@@ -49,7 +49,7 @@ namespace atlas
             isa_file_path_)),
         stop_sim_on_wfi_(p->stop_sim_on_wfi),
         hypervisor_enabled_(extension_manager_.isEnabled("h")),
-        vector_state_ptr_(new VectorState()),
+        vector_config_(new VectorConfig{}),
```

Comment:
Why is this a pointer?  Can you use a concrete object?  If you must use a pointer, you must use `unique_ptr`